### ISSUE TITLE
Logging framework

### ISSUE THREAD

**Issue by dmportella** [river-styx/river-styx.core#4]
Logging must be added.
should support logging to file
should support logging to custom endpoint (including queue - publishing)

**Comment by tegud** [comment]
Guess the real question is do we go with something like standard like winston, or specify an interface people can plug their own loggers into?

My preference would be the latter, it would feel a bit odd for a framework npm to enforce a choice of logging npm.

**Comment by tegud** [comment]
Joe makes a good point, maybe it's all events.

e.g. client.on('connection-made', (connectionDetails) => { // handler });

**Comment by dmportella** [comment]
yeah i was going to suggest that we should use events to to allow some one to hook it up it.

joe should join the conversation here

**Comment by dmportella** [comment]
@tegud i am also in the later group i rather make it so people can plug in their own logging...

**Comment by tegud** [comment]
So events it is, no logging needed

**Comment by dmportella** [comment]
aye that seems the best way just use this ticket to add the events if any is needed
